RELEASE_NOTES.txt

```diff
@@ -1,6 +1,372 @@
+* Fixed incorrect exception is thrown if no data nodes found for a partition
+* Fixed missing EntryProcessor resource injection
+* Fixed mvcc history can be missed during remove operation
+* Fixed mvcc-cache rebalance procedure leads to data loss
+* Fixed query trackers are not released
+* Fixed query with specified explicit partitions
+* Fixed race on invokeAll operations
+* Fixed read transactions remap when coordinator fails
+* Fixed removeAll on unstable topology
+* Fixed transaction asynchronous rollback
+* Fixed update operation hanging on backup on unstable topology
+* Fixed vacuums do not clean up all outdated versions
+
+
+JDBC:
+* Added cache expiry policies
+* Added support JDBC thin driver: connection timeout
+* Added support data page scan for JDBC
+* Added support of handling multiple async requests in parallel
+* Added support query cancel
+* Added support query timeout
+* Fixed JDBC getPrimaryKeys returns wrong value for COLUMN_NAME
+* Fixed JDBC thin driver metadata misses caches with queryEntities and names containing underscores
+* Fixed STREAMING mode fails on streamer reopen on not ordered mode
+* Fixed memory leak caused by executing a JDBC prepared statement
+* Fixed sqline command !tables works incorrectly for a client node
+* Fixed suspended optimistic transaction automatically resumes to last thread after a timeout
+* Now driver starts with a random server to connect
+* Updated handshake protocol so that the node returns its UUID
+
+
+ODBC:
+* Fixed SQLColumns does not work for tables with escape sequences in name
+* Fixed backward compatibility between 2.5 and 2.7
+
+
+Web Agent:
+* Added support for the cluster-ID
+* Added support of two way SSL authentication in Web Console agent
+* Fixed flag --disable-demo doesn't work
+* Fixed hostname verifier must be disabled if "-Dtrust.all=true".
+* Improve information message in case REST command failed on cluster
+* Improve logging when cluster topology changed
+
+
+Web Console:
+* Added TypeScript support
+* Added ability to cancel queries
+* Added ability to configure disk page compression properties
+* Added check for supported MongoDb version before WC start
+* Added email confirmation
+* Added management events (EVT_MANAGEMENT_TASK_STARTED)
+* Added missed configurations for some of the properties from IgniteConfiguration
+* Added option to disable self-registration on Web Console
+* Added possibility to configure distributed MVCC
+* Added support for "date", "time" and "date-and-time" on InputDialog
+* Added support for "type=number" on InputDialog
+* Added support for new countries Taiwan, Hong Kong, Singapore
+* Fixed can't select cluster memory eviction mode
+* Fixed long-running SQL queries could be randomly cancelled from WC
+* Fixed memory leak in chart component
+* Fixed priority for field validations
+* Fixed validation in Firefox
+* Update styles of top menu items
+* Update to RxJS 6
+
+
+Node.js:
+* Fixed Node.js thin client UUID marshalling
+
+
+Ignite .NET:
```

Comment:
I've reviewed Ignite .NET changelist, there are a few missing, and some are duplicate. I propose to list the following:

```
* Added Thin Client Partition Awareness
* Added Thin Client Expiry Policy
* Added Thin Client Cluster API
* Added .NET Core 3.x support
* Added Ignite.NET Dockerfile
* Added Runnable .NET Core assembly
* Added ICache.GetLongSize
* Added IClusterNode.Version
* Added ICompute.WithExecutor
* Added IIgnite.GetVersion
* Added IBinaryObjectBuilder.SetField(name, val, type) overload
* Added Partition Preload API (PreloadPartition, PreloadPartitionAsync)
* Fixed JVM thread leak
* Fixed serialization performance for ISerializable
```

Reply:
Thank you, Pavel.
I've fixed notes according your suggestions.